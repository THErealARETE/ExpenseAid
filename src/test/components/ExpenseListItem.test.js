import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expense from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';
Enzyme.configure({
    adapter: new Adapter()
})

test ('should render expense list item ', ()=>{
    const wrapper = shallow(<ExpenseListItem {...expense[0]}/>)
    expect(wrapper).toMatchSnapshot()
})
