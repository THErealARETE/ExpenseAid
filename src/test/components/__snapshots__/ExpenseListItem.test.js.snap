// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseListItem
    amount={200}
    createdAt={0}
    description="Gem"
    id="1"
    note=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/edit/1"
        >
          <h3>
            Gem
          </h3>
        </Link>,
        <p>
          200
           - 
          0
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h3>
            Gem
          </h3>,
          "replace": false,
          "to": "/edit/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gem",
          },
          "ref": null,
          "rendered": "Gem",
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            200,
            " - ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          200,
          " - ",
          0,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/edit/1"
          >
            <h3>
              Gem
            </h3>
          </Link>,
          <p>
            200
             - 
            0
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h3>
              Gem
            </h3>,
            "replace": false,
            "to": "/edit/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Gem",
            },
            "ref": null,
            "rendered": "Gem",
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              200,
              " - ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            200,
            " - ",
            0,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
